synapse/handlers/room_member.py

```diff
@@ -548,13 +549,17 @@ async def update_membership_locked(
         )
 
         if (
-            not self.allow_per_room_profiles and not is_requester_server_notices_user
-        ) or requester.shadow_banned:
+            (not self.allow_per_room_profiles and not is_requester_server_notices_user)
+            or requester.shadow_banned
+            or len(content.get("displayname", "")) > MAX_DISPLAYNAME_LEN
+        ):
             # Strip profile data, knowing that new profile data will be added to the
             # event's content in event_creation_handler.create_event() using the target's
             # global profile.
             content.pop("displayname", None)
             content.pop("avatar_url", None)
+            # content may now be emtpy
+            content_specified = len(content) == 0
```

Comment:
Why is this change necessary?

Reply:
It wasn't necessary, I'd just added it as I thought it was weird that content_specified might be incorrect. I've deleted it though as it wasn't really relevant to this